src/mon/OSDMonitor.cc

```diff
@@ -4551,6 +4894,74 @@ void OSDMonitor::tick()
   if (do_propose ||
       !pending_inc.new_pg_temp.empty())  // also propose if we adjusted pg_temp
     propose_pending();
+
+  {
+    std::lock_guard l(balancer_lock);
+    if (ceph_using_tcmalloc() && mon_memory_autotune && pcm != nullptr) {
+      pcm->tune_memory();
+      pcm->balance();
+      _set_new_cache_sizes();
+      dout(10) << "tick balancer "
+               << " inc cache_bytes: " << inc_cache->get_cache_bytes()
+               << " inc comtd_bytes: " << inc_cache->get_committed_size()
+               << " inc used_bytes: " << inc_cache->_get_used_bytes()
+               << " inc num_osdmaps: " << inc_cache->_get_num_osdmaps()
+               << dendl;
+      dout(10) << "tick balancer "
+               << " full cache_bytes: " << full_cache->get_cache_bytes()
+               << " full comtd_bytes: " << full_cache->get_committed_size()
+               << " full used_bytes: " << full_cache->_get_used_bytes()
+               << " full num_osdmaps: " << full_cache->_get_num_osdmaps()
+               << dendl;
+    }
+  }
+}
+
+void OSDMonitor::_set_new_cache_sizes()
+{
+  uint64_t cache_size = 0;
+  int64_t inc_alloc = 0;
+  int64_t full_alloc = 0;
+  int64_t kv_alloc = 0;
+
+  if (pcm != nullptr && rocksdb_binned_kv_cache != nullptr) {
+    cache_size = pcm->get_tuned_mem();
+    inc_alloc = inc_cache->get_committed_size();
+    full_alloc = full_cache->get_committed_size();
+    kv_alloc = rocksdb_binned_kv_cache->get_committed_size();
+  }
+
+  double avg_inc_bytes_per_osdmap = inc_cache->get_avg_bytes_per_osdmap();
+  double avg_full_bytes_per_osdmap = full_cache->get_avg_bytes_per_osdmap();
```

Comment:
why approximate when the cache knows how much space it uses now? it could implement a set_bytes() that trimmed until it was below the desired size

Reply:
It sounds like you are suggesting to include the monitoring of a new variable, for e.g.  `max_bytes`. The new `set_bytes()' function sets `max_bytes` and trims if necessary. `trim-cache()` now also needs to monitor the bytes usage and trims until it's below 'max-bytes`. 

This would mean `trim_cache()` be able to handle both LRU cache size and size in bytes. Please confirm
if this is what you have in mind.

Reply:
yes that's what I was thinking

Reply:
Sure, I have modified the SimpleLRU class. I had to add functions like `add_bytes()`, `_add_bytes()`
and `trim_cache_bytes()` to maintain backward compatibility with other users of this class. I had to do this since the original `add()` function also calls `trim_cache()`.

I initially tried to modify only the `trim_cache()` function by adding a condition for `max_bytes`, but realized that the cache size setting (`max_size`) will precede the bytes check as it's set to only 500 from the config options and this limit might likely be hit first.

If you think there's a better way, I can work on this over the weekend.

Reply:
that approach seems fine to me. certainly nothing to work on over the weekend!